In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df=pd.read_csv('Sales_add.csv')

In [3]:
df.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [4]:
df.isna().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

In [5]:
df.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

In [6]:
df.shape

(22, 5)

# The company wishes to clarify whether there is any increase in sales after stepping into digital marketing
# Null hypothes H0: sales after digital marketing <= sales before digital marketing
# Alternative hypothes H1: sales after digital marketing > sales before digital marketing

In [7]:
before=df['Sales_before_digital_add(in $)']
before

0     132921
1     149559
2     146278
3     152167
4     159525
5     137163
6     130625
7     131140
8     171259
9     141956
10    159339
11    178939
12    145062
13    151514
14    147463
15    177195
16    140862
17    167996
18    132135
19    152493
20    147425
21    130263
Name: Sales_before_digital_add(in $), dtype: int64

In [8]:
after=df['Sales_After_digital_add(in $)']
after

0     270390
1     223334
2     244243
3     231808
4     258402
5     256948
6     222106
7     230637
8     226261
9     193735
10    203175
11    276279
12    205180
13    253131
14    229336
15    187305
16    234908
17    191517
18    227040
19    212579
20    263388
21    243020
Name: Sales_After_digital_add(in $), dtype: int64

In [9]:
from scipy.stats import chi2_contingency
t_stat,p_value=stats.ttest_ind(after,before)
print('The value t-statistics is: ',t_stat)
print('The value p-value is: ',p_value)

The value t-statistics is:  12.995084451110877
The value p-value is:  2.614368006904645e-16


Assume alpha value = 0.05,
from results obtained above p-value is lesser than alpha value.
So reject null hypothesis.
Thus, we can conclude there is increase in sales after digital marketing

# The company needs to check whether there is any dependency between the features “Region” and “Manager”.

# Null hypothesis H0: Region and Manager are independent

# Alternative hypothesis H1: Region and Manager are dependent

In [10]:
cont=pd.crosstab(df['Region'],df['Manager'])
cont

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [11]:
stats.chi2_contingency(cont)

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

Assume alpha value=0.05, p-value=0.549, Thus p-value>alpha-value.
Accept Null Hypothesis.
THus, Region and Manager are independent.
